packages/calcite-components/src/components/input-date-picker/input-date-picker.tsx

```diff
@@ -893,6 +893,7 @@ export class InputDatePicker
     }
 
     if (key === "Enter") {
+      event.preventDefault();
```

Comment:
Something I'm noticing is that we're having to call `event.preventDefault()` for every key. This is fine, but maybe it would be good to have a dom utility function to handle this better for us.

Something like:

```ts
export function preventDefaultForKeys(event: KeyboardEvent, keys: string[]): void {
  if (!event.defaultPrevented && keys.includes(event.key)) {
    event.preventDefault();
  }
}
```

usage:

```ts
preventDefaultForKeys(["ArrowDown", "ArrowUp", "ArrowLeft", "ArrowRight"], [" ", "Enter"]);
```

what do you think @jcfranco?